pkg/k8s/cnp.go

```diff
@@ -78,7 +78,7 @@ func (c *CNPStatusUpdateContext) updateStatus(ctx context.Context, cnp *types.Sl
 		if option.Config.K8sEventHandover {
 			err = c.updateViaKVStore(ctx, cnp, false, false, policyImportErr, rev, cnp.Annotations)
 		} else {
-			err = c.updateViaAPIServer(cnp, false, false, policyImportErr, rev, cnp.Annotations)
+			err = c.UpdateViaAPIServer(cnp, false, false, policyImportErr, rev, cnp.Annotations)
```

Comment:
Why export the function if it's only used in unit tests?

Reply:
It's in the commit message. I had to the unit tests a dedicated package to avoid import cycle

Reply:
Why is there an import cycle? Is this the only way to avoid it?

Reply:
That's not what I was referring to. I meant to ask if you tried doing something like this to avoid exporting a method for unit tests only - https://medium.com/@robiplus/golang-trick-export-for-test-aa16cbd7b8cd.

Reply:
@christarazi the unit tests (`pkg/k8s`) imports `pkg/k8s/apis/cilium.io/v2/client` which imports `pkg/k8s`

Reply:
@aditighag Maybe I'm missing something but I don't see how that can work since the link referes to a package function where this is a structure's method?

Reply:
I think you'd need to have a test file specifically part of the `package k8s` which does `var UpdateViaAPIServer = updateViaAPIServer` which then you use in the `package k8s_test`. However, that's moot because of the import cycle problem. 

Do you think that we should refactor so we avoid the import cycle to begin with? I feel that when we run into an import cycle, that usually means we should reorg the code a bit, no?

Reply:
@christarazi but that won't work because `updateViaAPIServer` is a method, not a function, like in the example from the medium link.

>Do you think that we should refactor so we avoid the import cycle to begin with? I feel that when we run into an import cycle, that usually means we should reorg the code a bit, no?

I would say so but the import cycle happens in the testing package. I only found it when the CI complained as far as "production" code, it's not affected. I thought about on having a `benchmark` package inside `pkg/k8s` dedicated to this `_test.go` file but in the end it would be the same as simply renaming `package k8s` to `package k8s_test`, which is what the PR is doing.

Reply:
@aanm Sure, the trick works - https://github.com/aditighag/cilium/commit/aec66c0818aa06dabf0d0521c43d240c784e823f.

Reply:
@aditighag PTAL again.